snuba/state/rate_limit.py

```diff
@@ -162,12 +201,14 @@ def rate_limit(
             rate_limit_params.per_second_limit,
         ),
     ]
-
+    # see if we hit any rate limits
     reason = next((r for r in reasons if r.limit is not None and r.val > r.limit), None)
 
     if reason:
         try:
-            rds.zrem(bucket, query_id)  # not allowed / not counted
+            # when we remove the query because we rate limited it,
```

Comment:
I don't think you need to explain what the redis is doing, but it's worth commenting that we do it to prevent rate limited queries from counting towards the rate limit.